src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -302,7 +302,7 @@ void CarouselViewScrolled(object sender, ItemsViewScrolledEventArgs e)
 		[UnconditionalSuppressMessage("Memory", "MEM0003", Justification = "Proven safe in test: MemoryTests.HandlerDoesNotLeak")]
 		void CollectionViewUpdating(object sender, NotifyCollectionChangedEventArgs e)
 		{
-			if (ItemsView is not CarouselView carousel)
+			if (ItemsSource.ItemCount == 0 || ItemsView is not CarouselView carousel)
```

Comment:
Could you include an UITest?

Reply:
> Could you include an UITest?

Hi @jsuarezruiz 
The test case for this scenario is already present. I have updated it to use the default CarouselView instead of CarouselView2.